lib/internal/socket_list.js

```diff
@@ -1,104 +1,105 @@
 'use strict';
 
-module.exports = {SocketListSend, SocketListReceive};
-
 const EventEmitter = require('events');
-const util = require('util');
 
 // This object keep track of the socket there are sended
-function SocketListSend(slave, key) {
-  EventEmitter.call(this);
+class SocketListSend extends EventEmitter {
+  constructor(slave, key) {
+    super();
+    this.key = key;
+    this.slave = slave;
```

Comment:
Likewise, it would be awesome if we could change this during the refactor (here and in the other parts of the file this PR touches).